force-app/main/default/classes/ERR_Notifier_TEST.cls

```diff
@@ -0,0 +1,335 @@
+        Test.stopTest();
+
+        System.assertEquals(
+            ERR_Notifier.instance,
+            notifierService,
+            'Instance of service class returned should match static instance variable.'
+        );
+        System.assertEquals(true, notifierService != null, 'Instance of service class should not be null.');
+        System.assertEquals(true, notifierService.sendEmailAsync, 'sendEmailAsync value should be true');
+    }
+
+    /**************************************************************************************************************************
+     * @description Test method to verify that the getInstance method returns the existing instance of the
+     * ERR_Notifier class when one already exists.
+     ***************************************************************************************************************************/
+    @isTest
+    private static void getInstanceExisting() {
+        Test.startTest();
+        ERR_Notifier notifierServiceInstance1 = ERR_Notifier.getInstance();
+
+        ERR_Notifier notifierServiceInstance2 = ERR_Notifier.getInstance();
+        Test.stopTest();
+
+        System.assertEquals(
+            ERR_Notifier.instance,
+            notifierServiceInstance1,
+            'Instance of service class returned should match static instance variable.'
+        );
+
+        System.assertEquals(
+            notifierServiceInstance1,
+            notifierServiceInstance2,
+            'Subsequent retrievals of service class instance should return existing instance.'
+        );
+
+        System.assertEquals(true, notifierServiceInstance1 != null, 'Instance of service class should not be null.');
+    }
+
+    /**********************************************************************************************
+     * @description Test retrieval of System Administrator Users' Emails
+     * verify: Emails of System Administrator Users are retrieved
+     */
+    @IsTest
+    private static void shouldMatchSystemAdminEmailList() {
+        ERR_Notifier notifierService = ERR_Notifier.getInstance();
+        Test.startTest();
+        Set<String> recipientEmails = new Set<String>(
+            notifierService.getErrorEmailRecipients(ERR_Notifier.notificationOptions.sysAdmins)
+        );
+        Test.stopTest();
+
+        Set<String> expectedEmails = getSystemAdminEmails();
+
+        System.assertEquals(expectedEmails.size(), recipientEmails.size(), 'Number of emails should match');
+        System.assert(expectedEmails.containsAll(recipientEmails), 'Expected and retrieved emails should be the same');
+    }
+
+    /**********************************************************************************************
+     * @description Confirm that the method to retrieve Error log record pending notification
+     * only retrieve records created in the last 48 hours.
+     */
+    @IsTest
+    private static void shouldOnlyRetrieveRecentErrorLogEntries() {
```

Comment:
Would it be worth splitting this into two separate tests: one to assert against max age (where it doesn't reach max records) and the other to assert max records (which this one is doing)? For this one, it isn't necessarily testing max age as the max records gets hit first.

Reply:
Makes sense. Added 2 different test methods to test record count and age limits.